pipeline/src/cache_table.rs

```diff
@@ -0,0 +1,658 @@
+//! [`CacheTable`] for querying `Vec<RecordBatch>` by DataFusion.
```

Comment:
FYI, this is a copy of the MemTable implementation that allows for clearing all the data in the table. We could upstream that change however I think we will likely end up optimizing this implementation over time and so having our own copy as a starting point is useful.